tests/test_urlgrabber.py

```diff
@@ -158,3 +158,43 @@ def test_http_not_found(self):
             self._runner(self.httpbin.url + "/status/404", 404)
         with pytest.raises(urllib.error.HTTPError):
             self._runner(self.httpbin.url + "/no/such/file", 404)
+
+
+class TestFilenameFromDispositionHeader:
+    @pytest.mark.parametrize(
+        "header, result",
+        [
+            (
+                # In this case the first filename (not the UTF-8 encoded) is parsed.
+                "filename=Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb; filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD-utf.nzb",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb"
+            ),
+            (
+                "filename=Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb;",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb"
+            ),
+            (
+                "filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb",
+                "Zombie.Land.Saga.Revenge.S02E12.480p.x264-mSD.nzb"
+            ),
+            (
+                "attachment; filename=jakubroztocil-httpie-0.4.1-20-g40bd8f6.tar.gz",
```

Comment:
Supplying additionally `attachment` here, it's added by the function later.

Reply:
A real example of what's being passed to the function is: `attachment; filename=Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb; filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb`. So that also has the `attachment` portion in it?

Or maybe i'm misunderstanding you?

Reply:
Check this line: https://github.com/sabnzbd/sabnzbd/pull/1946/files#diff-33d9efe9612e1d7ed61b9b8cdff238927ced12175411df050f92d3492912be34R75
In the function itself you add the `attachment; ` part, so no need to add to the input in the tests here.
You also don't add it a few tests-up: https://github.com/sabnzbd/sabnzbd/pull/1946/files#diff-3aa82f6816c2807e5b3fbd07c7c10c7affb12f4dcbb81988decaa492dcbf7dfbR178

Reply:
I just wanted to have a test resemble  a real situation where `attachment` is also passed. 
Like with the prowlarr example. Attachment is included a the string.

Reply:
@Safihre you've asked me to make this change. And i've added `attachment; ` here. 
```python
Message(f'Content-Disposition: attachment; {content_disposition}')
```

But when debugging i notice that it will effectively parse: 
```
Message(f'Content-Disposition: attachment; attachment; filename=Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb; filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb')
```

Now this isn't that much of an issue. As also proven by my tests, it will get the filename from it. So as prowlarr is sending a content-disposition header conform RFC like: `attachment; filename=Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb; filename*=UTF-8''Zombie.Land.Saga.Revenge.S02E12.1080p.WEB.H264-SENPAI.nzb` I'd also like to have something like this in one of the tests.